### PR TITLE
feat: add set and fetch relation tests

### PR DESCRIPTION
**Pull request by richtia**

### CODE REVIEW

File: substrait_consumer/producers/datafusion_producer.py [hunk 1]
```diff
@@ -42,7 +42,11 @@ def produce_substrait(self, sql_query: str, validate = False, ibis_expr: str = N
         substrait_plan = ss.serde.serialize_to_plan(sql_query, self._ctx)
         substrait_plan_bytes = substrait_plan.encode()
         if validate:
-            sv.check_plan_valid(substrait_plan_bytes)
+            config = sv.Config()
+            config.override_diagnostic_level(1002, "warning", "info")  # error
+            config.override_diagnostic_level(7, "warning", "info") # warning
```
[the remark this answers is not in the document]
Comment (richtia) [reply substrait_consumer/producers/datafusion_producer.py:47]: I'll add more details in the comments.  This one is
```
ValueError: Warning at plan.version: cannot automatically determine whether plan version (0.51.0) is compatible with the Substrait version that this version of the validator validates (0.20.0). Please check the release notes between these versions, or install the correct version of the validator. See also https://github.com/substrait-io/substrait/pull/210#discussion_r881965837 (code 0007)
```

File: substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+              ]
+            },
+            "projection": {
+              "select": {
+                "structItems": [
+                  {}
+                ]
+              }
+            },
+            "namedTable": {
```
[the remark this answers is not in the document]
Comment (richtia) [reply substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json:27]: Yeah, the read relations are going in another PR.  I was categorizing the different relations/operations here: https://substrait.io/relations/logical_relations/#fetch-operation

File: substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json
[the remark this answers is not in the document]
Comment (richtia) [reply substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json:1]: Based on the SQL, I was expecting something more like this using a fetch: https://github.com/substrait-io/consumer-testing/pull/100/files/5d23304625eb84f8d1c9dc6095d4dc98ce585c76#diff-4734c8e5fda2a30579d19368aa9ef21a12f67d9c9f42f1b970cf11a4bd2bfa1a

How do you think we should handle this situation?

File: substrait_consumer/tests/functional/relations/set_relation_snapshots/IsthmusProducer/union_all_plan.json
[the remark this answers is not in the document]
Comment (richtia) [reply substrait_consumer/tests/functional/relations/set_relation_snapshots/IsthmusProducer/union_all_plan.json:1]: the snapshot is under the `set_relation_snapshots` directory already, but i can add `set_` to the test case name too if you think that's better?

[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply substrait_consumer/tests/functional/relations/set_relation_snapshots/IsthmusProducer/union_all_plan.json:1]: Hmm, no need to duplicate the names (that's more of a Java style thing).  We should be consistent with whatever approach we take though.

File: substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply substrait_consumer/tests/functional/relations/fetch_relation_snapshots/DataFusionProducer/fetch_plan.json:1]: It's hard to get all producers to convert SQL the same way.  We can leave it.

**Review by EpsilonPrime — APPROVED** [review]

**Pull request merged**